elastic/kibana
# [Security Solution][Rules] Allow add_prepackaged_rules to change rule types

## Summary

Closes https://github.com/elastic/kibana/issues/123859

Adds support for rule type changes in when updating prepackaged rules, since some prepackaged rules have already changed types in the past. The old rule is first deleted, then the new rule is created with the same `id` as the old rule and the `enabled`, `actions`, and `exceptions_list` settings are carried over to the new rule.

Does *not* add support for changing rule types through other routes, e.g. update, import, patch.

Testing:
1. Download the ES data at https://upload.elastic.co/d/715195675ea6bbb3e7942667a5bf8ad497f751c05ee00d68733121bfe1b1c0e2 - this is a 7.17 ES instance with prebuilt rules from 7.13 installed.
2. Unzip the ES data into a folder and start ES on this branch using `yarn es snapshot -E path.data=<FOLDER_WITH_ES_DATA> -E xpack.security.authc.api_key.enabled=true --license trial`
3. Start Kibana with `yarn start`
4. Visit the rules management page. You should see all the prebuilt rules installed and a message saying prebuilt rules can be updated.
5. `Creation of Hidden Files and Directories` is the prebuilt rule that will change types when you update the prebuilt rules. Make any test changes to the configuration before updating the rules, e.g. add actions, exceptions, enable/disable the rule.
6. Update the prebuilt rules. Verify that configuration settings on the rule are retained and `Creation of Hidden Files and Directories` has been converted to an EQL rule.
